ssl/quic/quic_rstream.c

```diff
@@ -135,3 +184,96 @@ int ossl_quic_rstream_available(QUIC_RSTREAM *qrs, size_t *avail, int *fin)
 #endif
     return 1;
 }
+
+int ossl_quic_rstream_get_record(QUIC_RSTREAM *qrs,
+                                 const unsigned char **record, size_t *rec_len,
+                                 int *fin)
+{
+    const unsigned char *record_ = NULL;
+    size_t rec_len_, max_len;
+
+    if (!ossl_sframe_list_lock_head(&qrs->fl, &qrs->head_range, &record_, fin)) {
+        *record = NULL;
+        *rec_len = 0;
+        return 0;
```

Comment:
Is this actually an error? Doesn't this just mean there is no data to be read? I wonder if this should be reported as "we successfully read 0 bytes" - we need to be able to distinguish this from "real" failures such as from the ossl_assert below.